src/parser/mage/fire/modules/features/Combustion.tsx

```diff
@@ -0,0 +1,267 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from 'common/format';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import { formatNumber } from 'common/format';
+import Events from 'parser/core/Events';
+import SpellUsable from 'parser/shared/modules/SpellUsable';
+import AbilityTracker from 'parser/shared/modules/AbilityTracker';
+import SUGGESTION_IMPORTANCE from 'parser/core/ISSUE_IMPORTANCE';
+
+const debug = false;
+const FIRESTARTER_HEALTH_THRESHOLD = .90;
+const DAMAGE_SPELLS = [
+  SPELLS.FIREBALL,
+  SPELLS.PYROBLAST,
+  SPELLS.SCORCH,
+  SPELLS.FIRE_BLAST,
+  SPELLS.PHOENIX_FLAMES_TALENT,
+];
+
+class Combustion extends Analyzer {
+  static dependencies = {
+    spellUsable: SpellUsable,
+    abilityTracker: AbilityTracker,
+  };
+  protected spellUsable!: SpellUsable;
+  protected abilityTracker!: AbilityTracker;
+
+  hasFlameOn: boolean; hasPhoenixFlames: boolean; hasFirestarter: boolean;
+  castEvent!: {
```

Comment:
This should not have the !, but its value should have | undefined appended.

Reply:
Looking at your usage, it should probably be `CastEvent | undefined`

Reply:
I think i tried that but then flagTimeline() complained that castEvent could be undefined. I told VS Code to auto fix it and that is what it did. I can try and change it back and see if it works though.

Reply:
How about making the event a part of the parameters of `flagTimeline`? Then TS will see that you always take care to ensure the property has a value and pass.

Reply:
I think i tried that too, but i might have just thought about that and decided to let vs code fix it instead, not sure